### ISSUE TITLE
Chat dialog opens when I play a game - annoying

### ISSUE THREAD

**Issue by pychess** [pychess/pychess#571]
Original [issue 571](https://code.google.com/p/pychess/issues/detail?id=571) reported by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-07-12

**What steps will reproduce the problem?**
When I start to play sometimes opponent writes "Hi" etc.
What I don't like is that chat window is opened over existing chess-board window. This can be annoyed, because I don't have a time to chat when I only have 5 minutes to play. So this chat window is annoying.

**What is the expected output? What do you see instead?**
I think that instead of chat window any kind of chat that is related to the current game should appear in the chat tab (see attached picture).

**Did you use an installed version of PyChess or did you run in from a**
**tarball/svn?**
Current hg.

**Comment by pychess** [comment]
- attached [chat.png](https://code.google.com/p/pychess/issues/detail?id=571#c0)

**Comment by pychess** [comment]
Original [comment 1.](https://code.google.com/p/pychess/issues/detail?id=571#c1) posted by [gbtami](https://code.google.com/u/gbtami/) 2010-07-13
- changed summary to: Chat dialog opens when I play a game - annoying

**Comment by pychess** [comment]
Original [comment 2.](https://code.google.com/p/pychess/issues/detail?id=571#c2) posted by [lobais](https://code.google.com/u/lobais/) 2010-07-18

Agreed that it is annoying.
However the view next to the game, is currently only for chatting with your opponent.
Perhaps the chat window should just blink in the taskbar?
- added label: Post-0.10
- added label: Enhancement

**Comment by pychess** [comment]
Original [comment 3.](https://code.google.com/p/pychess/issues/detail?id=571#c3) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-07-19

"Perhaps the chat window should just blink in the taskbar?" - this is much better.

But what I see is that a lot of time (I am almost convinced that all of the time) I get new window from my opponent, so that is why I have suggested to just use Chat tab.

**Comment by pychess** [comment]
Original [comment 4.](https://code.google.com/p/pychess/issues/detail?id=571#c4) posted by [lobais](https://code.google.com/u/lobais/) 2010-07-19

Right. That's more critical.
Perhaps the opponent isn't using the right chat command. But we should be able to drag the conversation in anyway.
- added label: Medium

**Comment by pychess** [comment]
Original [comment 5.](https://code.google.com/p/pychess/issues/detail?id=571#c5) posted by [lobais](https://code.google.com/u/lobais/) 2013-06-27

I chased this to the fact that BoardManager uses the regex `names = "(\w+)"` and ChatManager uses `names = "([A-Za-z]+)"+titles`. This means that when ICPlayer tests:

```
def \_\_onPrivateMessage (self, cm, name, title, isadmin, text):
    if name == self.name:
        self.emit("offer", Offer(CHAT\_ACTION, param=text))
```

It has name='GuestABCD' and self.name='GuestABCD  (U)' which means it doesn't catch it, and the message is never propegated ICPlayer -> ICGameModel -> Human -> ChatPanel.

It would be nice to not have regexes like 'name' redefined in each icManager, but at this late hour I don't feel 100% certain that it isn't necessary in some conditions. If anybody more recently into FICS can help me out, I'd be grateful.

**Comment by pychess** [comment]
Original [comment 6.](https://code.google.com/p/pychess/issues/detail?id=571#c6) posted by [gbtami](https://code.google.com/u/gbtami/) 2013-07-08

Seems the private message regexp in chat manager is incorrect. There is no gameno part in private messages at all.
Instead of
                "%s(*)?(?:[(\d+)])? (?:tells you|says): (._)" % names)
we should use
                "%s(\_)? (?:tells you|says): (.*)" % names)

**Comment by pychess** [comment]
Original [comment 7.](https://code.google.com/p/pychess/issues/detail?id=571#c7) posted by [mattgatto](https://code.google.com/u/mattgatto/) 2013-07-19

I moved alot of stuff into ic/__init__.py when I merged FICSObjects.py and other stuff, but I left alot of these regexps like 'name' alone for reasons I can't remember. I wouldn't move these unless you're going to do alot of testing and/or write test cases for players with different types of titles in their names in all the major areas of the FICS Lounge. It should be done eventually though. Nice catch BTW. This bug was pretty annoying.

**Comment by pychess** [comment]
Original [comment 8.](https://code.google.com/p/pychess/issues/detail?id=571#c8) posted by [gbtami](https://code.google.com/u/gbtami/) 2013-07-20

Fixed comment 5. and comment 6.. Now chat messages typed by 1. player in hes chat side panel appears in 2. player's console window immediately, but I have to wait looong time to appear in 2. players chat side panel. Strange.

**Comment by pychess** [comment]
Original [comment 9.](https://code.google.com/p/pychess/issues/detail?id=571#c9) posted by [gbtami](https://code.google.com/u/gbtami/) 2013-07-21

This issue was closed by revision 6077249f6e40.
- changed status to: Fixed
